<!DOCTYPE html><!--[if lt IE 8]><html lang="en" class="ie"><![endif]--><!--[if gt IE 8]><!--><html lang="en"><!--<![endif]--><head><meta charset="utf-8"><title>MOLGENIS - Python</title><meta name="description" content="Access via Python"><meta name="author" content="MOLGENIS community"><meta name="handheldfriendly" content="true"><meta name="mobileoptimized" content="320"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"><meta http-equiv="cleartype" content="on"><script src="/res/js/jquery-1.11.3.min.js"> </script><script src="/res/js/bootstrap.min.js"></script><link rel="stylesheet" href="/res/css/prism.css"><link rel="stylesheet" href="/res/css/application.css" type="text/css"><link rel="shortcut icon" href="/favicon.png" type="image/png"><link rel="alternate" type="application/rss+xml" href="/blog/feed.xml" title="undefined"></head><body><header class="molgenis-docs-header"><div role="navigation" class="navbar navbar-fixed-top"><div class="header-container"><div class="navbar-header"><button type="button" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar" class="navbar-toggle collapsed"><span class="sr-only">Toggle navigation</span><span class="icon-bar"></span><span class="icon-bar"></span><span class="icon-bar"></span></button><a href="/" class="navbar-brand active"><img src="/res/images/wordmark-blue.png" alt="undefined" class="wordmark"/></a></div><div class="collapse navbar-collapse">              <ul class="nav navbar-nav"><li><a href="/software">TOOLS</a></li><li class="active"><a href="/documentation">DOCUMENTATION</a></li><li><a href="/pipelines">PIPELINES</a></li><li><a href="/develop">DEVELOP</a></li><li><a href="/support">SUPPORT</a></li><li><a href="#searchbox" data-toggle="collapse"><i class="glyphicon glyphicon-search"></i></a></li></ul></div></div></div><div id="searchbox" class="collapse"><form role="search" action="/search-results"><input placeholder="Search" name="q" class="form-control"/></form></div><div class="molgenis-docs-titlebar active"><div class="container"> <h1>Python</h1><p>Access via Python   </p></div></div></header><div class="molgenis-docs-body"><div class="container"><div class="row"><div role="complementary" class="col-md-3"><nav id="molgenis-docs-sidebar" class="molgenis-docs-sidebar hidden-print hidden-xs hidden-sm"><ul id="molgenis-docs-sidenav" class="nav molgenis-docs-sidenav"></ul></nav><script>$(function() {
    var sideNav = $("#molgenis-docs-sidenav");
    var index = 0;
    $(".row h1").each(function() {
        
        //create id for h1
        var id = (index++) + $(this).text().replace(/\s/gi,"-").replace(/\W/g,"").toLowerCase();
        $(this).prepend("<a name=\""+id+"\"></a>");
        
        //add h1 to list
        var li = $("<li><a href=\"#"+id+"\">"+$(this).text()+"</a></li>")
        sideNav.append(li);
        
        //wrap h1 into section
        var section = $(this).wrap("<section id=\""+id+"\"></section>").parent();
        
        //wrap parts of h1 into one section
        $(this).parent().nextUntil("h1").each(function(){            
            section.append($(this));
        });
        
        //repeat for h2
        if($("h2",$(this).parent()).length > 0) {
            var ul = $("<ul class=\"nav\"></ul>");
            li.append(ul);
            
            $("h2",$(this).parent()).each(function(){
            
                //create id for h2
                var id = (index++) + $(this).text().replace(/\s/gi,"-").replace(/\W/g,"").toLowerCase();
                $(this).prepend("<a name=\""+id+"\"></a>");
        
                //wrap h2 into section
                var section = $(this).wrap("<section id=\""+id+"\"></section>").parent();
        
                ul.append($("<li><a href=\"#"+id+"\">"+$(this).text()+"</a></li>"));
        
                //wrap parts of h2 into one section
                $(this).parent().nextUntil("h2").each(function(){            
                    section.append($(this));
                });
            

            });
        }
    });    
    
    //add "back to top"
    sideNav.append("<li><a href=\"#top\" style=\"font-size:smaller\">Back to top</a></li>");
    var $body   = $(document.body);
    
    //enable the 'active' highlighting using scrollspy plugin
    $('body').scrollspy({
        target: '#molgenis-docs-sidebar'
    });
    $('#molgenis-docs-sidebar').width($('#molgenis-docs-sidebar').parent().width());
    
    //enable affix
    $('#molgenis-docs-sidebar').affix({
        offset: {
            top: $('#molgenis-docs-sidebar').offset().top,
            bottom: $('.molgenis-docs-footer').outerHeight(true) + 10
        }
    });
    
    //resize affix box when window resizes
    $(window).resize(function () {
            $('.molgenis-docs-sidebar').width($('#molgenis-docs-sidebar').parent().width());
        });
    
    $("table").addClass("table table-bordered");
});</script></div><div role="main" class="col-md-9 contents"><p><strong>
The Python API is a script that allows access to your MOLGENIS data from python. It is available on every MOLGENIS with version 1.14.0 and up on the URL <code>http[s]://&lt;your.molgenis.url&gt;/molgenis.py</code>.
</strong></p>
<h1>Example</h1><p>As an example, let&#39;s create a plot for publicly available ASE data available on <a href="https://molgenis56.target.rug.nl/">https://molgenis56.target.rug.nl/</a>. For a description of the data, take a look at <a href="http://molgenis.org/ase">http://molgenis.org/ase</a>.</p>
<p>We&#39;ll be creating a scatter plot so if you haven&#39;t already, install matplotlib from the commandline:</p>
<pre><code class="language-sh">pip install matplotlib
</code></pre>
<p>Download the python api from a molgenis server, for instance <a href="https://molgenis01.target.rug.nl/molgenis.py">https://molgenis01.target.rug.nl/molgenis.py</a> and save it in <code>molgenis.py</code>.</p>
<p>Start an interactive python shell and create a molgenis connection:</p>
<pre><code class="language-python">import molgenis
</code></pre>
<p>This imports the molgenis package.</p>
<pre><code class="language-python">session = molgenis.Session(&quot;https://molgenis56.target.rug.nl/api/&quot;)
</code></pre>
<p>Instantiates a new <code>Session</code> pointing at the molgenis56 server. If you take a look at the connection by typing</p>
<pre><code class="language-python">dir(session)
</code></pre>
<p>you should see, amongst others, the methods you can call:</p>
<pre><code class="language-python">[...,&#39;add&#39;, &#39;add_all&#39;, &#39;delete&#39;, &#39;get&#39;, &#39;get_attribute_meta_data&#39;, &#39;get_entity_meta_data&#39;, &#39;login&#39;, &#39;logout&#39;,...]
</code></pre>
<p>Let&#39;s load some data from the server using <code>session.get</code>:</p>
<pre><code class="language-python">session.get(&quot;ASE&quot;)
</code></pre>
<p>This retrieves the top 1000 rows from the ASE entity.</p>
<pre><code class="language-python">[{u&#39;Alternative_allele&#39;: u&#39;A&#39;, u&#39;P_Value&#39;: 2.06504739339637e-17, u&#39;Genes&#39;: {u&#39;href&#39;: u&#39;/api/v1/ASE/rs9901673/Genes&#39;}, u&#39;Fraction_alternative_allele&#39;: 0.479, u&#39;Pos&#39;: 7484101, u&#39;Reference_allele&#39;: u&#39;C&#39;, u&#39;Chr&#39;: u&#39;17&#39;, u&#39;href&#39;: u&#39;/api/v1/ASE/rs9901673&#39;, u&#39;Samples&#39;: u&#39;145&#39;, u&#39;Likelihood_ratio_test_D&#39;: 72.0813644150712, u&#39;SNP_ID&#39;: u&#39;rs9901673&#39;}, {u&#39;Alternative_allele&#39;: u&#39;T&#39;, u&#39;P_Value&#39;: 8.78109735398113e-18, u&#39;Genes&#39;: {u&#39;href&#39;: u&#39;/api/v1/ASE/rs2597775/Genes&#39;}, u&#39;Fraction_alternative_allele&#39;: 0.479, u&#39;Pos&#39;: 17503382, u&#39;Reference_allele&#39;: u&#39;C&#39;, u&#39;Chr&#39;: u&#39;4&#39;, u&#39;href&#39;: u&#39;/api/v1/ASE/rs2597775&#39;, u&#39;Samples&#39;: u&#39;359&#39;, u&#39;Likelihood_ratio_test_D&#39;: 73.769089117417, u&#39;SNP_ID&#39;: u&#39;rs2597775&#39;}, {u&#39;Alternative_allele&#39;: u&#39;C&#39;, u&#39;P_Value&#39;: 1.4917458949834e-18, u&#39;Genes&#39;: {u&#39;href&#39;: u&#39;/api/v1/ASE/rs3216/Genes&#39;}, u&#39;Fraction_alternative_allele&#39;: 0.479, u&#39;Pos&#39;: 214421, u&#39;Reference_allele&#39;: u&#39;G&#39;, u&#39;Chr&#39;: u&#39;11&#39;, u&#39;href&#39;: u&#39;/api/v1/ASE/rs3216&#39;, u&#39;Samples&#39;: u&#39;301&#39;, u&#39;Likelihood_ratio_test_D&#39;: 77.2691957930797, u&#39;SNP_ID&#39;: u&#39;rs3216&#39;}, [...],{u&#39;Alternative_allele&#39;: u&#39;T&#39;, u&#39;P_Value&#39;: 0.000132500824069775, u&#39;Genes&#39;: {u&#39;href&#39;: u&#39;/api/v1/ASE/rs1056019/Genes&#39;}, u&#39;Fraction_alternative_allele&#39;: 0.482, u&#39;Pos&#39;: 41337435, u&#39;Reference_allele&#39;: u&#39;C&#39;, u&#39;Chr&#39;: u&#39;12&#39;, u&#39;href&#39;: u&#39;/api/v1/ASE/rs1056019&#39;, u&#39;Samples&#39;: u&#39;47&#39;, u&#39;Likelihood_ratio_test_D&#39;: 14.605874945467, u&#39;SNP_ID&#39;: u&#39;rs1056019&#39;}]
</code></pre>
<p>Let&#39;s retrieve a specific SNP from the ASE entity:</p>
<pre><code class="language-python">print(session.get(&quot;ASE&quot;, q=[{&quot;field&quot;:&quot;SNP_ID&quot;, &quot;operator&quot;:&quot;EQUALS&quot;, &quot;value&quot;:&quot;rs12460890&quot;}]))
</code></pre>
<pre><code class="language-python">[{u&#39;Alternative_allele&#39;: u&#39;T&#39;, u&#39;P_Value&#39;: 7.1708540619282e-14, u&#39;Genes&#39;: {u&#39;href&#39;: u&#39;/api/v1/ASE/rs12460890/Genes&#39;}, u&#39;Fraction_alternative_allele&#39;: 0.527, u&#39;Pos&#39;: 829568, u&#39;Reference_allele&#39;: u&#39;C&#39;, u&#39;Chr&#39;: u&#39;19&#39;, u&#39;href&#39;: u&#39;/api/v1/ASE/rs12460890&#39;, u&#39;Samples&#39;: u&#39;21&#39;, u&#39;Likelihood_ratio_test_D&#39;: 56.0207947348388, u&#39;SNP_ID&#39;: u&#39;rs12460890&#39;}]
</code></pre>
<p>This SNP has a mild but significant allele-specific expression, based on expression counts in 21 samples.</p>
<p>Let&#39;s retrieve the samples for this SNP:</p>
<pre><code class="language-python">samples = session.get(&quot;SampleAse&quot;, q=[{&quot;field&quot;:&quot;SNP_ID&quot;, &quot;operator&quot;:&quot;EQUALS&quot;, &quot;value&quot;:&quot;rs12460890&quot;}])
print(samples)
</code></pre>
<pre><code class="language-python">[{u&#39;Ref_Counts&#39;: u&#39;130&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418785&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418785/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418785/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;121&#39;, u&#39;ID&#39;: u&#39;1418785&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;4142&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418786&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418786/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418786/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;4791&#39;, u&#39;ID&#39;: u&#39;1418786&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;19&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418787&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418787/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418787/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;28&#39;, u&#39;ID&#39;: u&#39;1418787&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;19&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418788&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418788/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418788/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;23&#39;, u&#39;ID&#39;: u&#39;1418788&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;32&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418789&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418789/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418789/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;11&#39;, u&#39;ID&#39;: u&#39;1418789&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;639&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418790&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418790/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418790/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;572&#39;, u&#39;ID&#39;: u&#39;1418790&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;202&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418791&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418791/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418791/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;309&#39;, u&#39;ID&#39;: u&#39;1418791&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;423&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418792&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418792/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418792/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;401&#39;, u&#39;ID&#39;: u&#39;1418792&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;271&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418793&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418793/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418793/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;234&#39;, u&#39;ID&#39;: u&#39;1418793&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;806&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418794&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418794/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418794/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;1081&#39;, u&#39;ID&#39;: u&#39;1418794&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;213&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418795&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418795/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418795/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;201&#39;, u&#39;ID&#39;: u&#39;1418795&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;74&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418796&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418796/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418796/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;96&#39;, u&#39;ID&#39;: u&#39;1418796&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;730&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418797&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418797/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418797/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;655&#39;, u&#39;ID&#39;: u&#39;1418797&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;584&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418798&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418798/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418798/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;699&#39;, u&#39;ID&#39;: u&#39;1418798&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;331&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418799&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418799/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418799/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;391&#39;, u&#39;ID&#39;: u&#39;1418799&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;13&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418800&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418800/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418800/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;14&#39;, u&#39;ID&#39;: u&#39;1418800&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;70&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418801&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418801/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418801/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;101&#39;, u&#39;ID&#39;: u&#39;1418801&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;47&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418802&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418802/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418802/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;35&#39;, u&#39;ID&#39;: u&#39;1418802&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;19&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418803&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418803/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418803/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;28&#39;, u&#39;ID&#39;: u&#39;1418803&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;44&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418804&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418804/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418804/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;47&#39;, u&#39;ID&#39;: u&#39;1418804&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;}, {u&#39;Ref_Counts&#39;: u&#39;60&#39;, u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418805&#39;, u&#39;SampleIds&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418805/SampleIds&#39;}, u&#39;Position&#39;: 829568, u&#39;SNP_ID&#39;: {u&#39;href&#39;: u&#39;/api/v1/SampleAse/1418805/SNP_ID&#39;}, u&#39;Alt_Counts&#39;: u&#39;55&#39;, u&#39;ID&#39;: u&#39;1418805&#39;, u&#39;Chromosome&#39;: u&#39;19&#39;
}]
</code></pre>
<p>There they are.</p>
<p>Let&#39;s format the expression counts</p>
<pre><code class="language-python">for sample in samples:
    print(&quot;{Ref_Counts:5} {Alt_Counts:5}&quot;.format(**sample))
</code></pre>
<pre><code class="language-python">130      121
4142     4791
19       28
19       23
32       11
639      572
202      309
423      401
271      234
806      1081
213      201
74       96
730      655
584      699
331      391
13       14
70       101
47       35
19       28
44       47
60       55
</code></pre>
<p>Let&#39;s plot the expression counts in these samples in a scatter plot.</p>
<pre><code class="language-python">import matplotlib.pyplot as plt
plt.scatter([sample[&quot;Ref_Counts&quot;] for sample in samples], [sample[&quot;Alt_Counts&quot;] for sample in samples])
plt.xlim([0, 5000])
plt.ylim([0, 5000])
plt.xlabel(&quot;Reference Allele&quot;)
plt.ylabel(&quot;Alternative Allele&quot;)
plt.title(&quot;Allele-Specific Expression for rs12460890&quot;)
</code></pre>
<p>And add a line for the non-specific expression.</p>
<pre><code class="language-python">plt.plot([0, 5000], [0, 5000])
plt.show()
</code></pre>
<p><img src="../res/images/rs12460890-py.png" alt="image"></p><center><br><i>See something that could be better? <a href="https://github.com/molgenis/molgenis.org/tree/master/public/documentation/guide-python.md">Edit this page at Github</a></i><br></center></div></div></div></div><footer class="molgenis-docs-footer"><p>MOLGENIS and these docs are maintained by the <a href="http://www.gccgroningen.nl/about-us/people-at-the-gcc/">MOLGENIS team and collaborators</a>. <br/> Documentation source code released under the <a href="/LICENSE-MIT">MIT License</a>, documentation contents under <a href="/LICENSE-CC">CC BY 3.0</a>, and MOLGENIS software code under the <a href="/LICENSE-LGPLv3">LGPLv3 License</a>.<ul class="text-muted"><li>Version 1.8.0</li><li>·</li><li><a href="https://github.com/molgenis/molgenis">MOLGENIS Github</a></li><li>·</li><li><a href="http://github.com/molgenis/molgenis/issues">MOLGENIS Issues</a></li><li>·</li><li><a href="https://github.com/molgenis/molgenis.org/tree/master/public">Docs Github    </a></li><li>·</li><li><a href="http://github.com/molgenis/molgenis.org/issues">Docs Issues</a></li><li>·</li><li><a href="https://github.com/molgenis/molgenis/releases">Releases</a></li></ul></p></footer><script src="/res/js/prism.js"></script></body></html>